### PR TITLE
Add Imix board definition

### PR DESCRIPTION
**Pull request by alevy**
Basic board definition including:

  * Console, gpio and timer drivers

  * Process loading (done the same as firestorm)

  * Flashing the kernel with openocd

  * Flashing apps with openocd (multiple apps supported)

@shaneleonard can you take a look at the pin configurations? I copied from Firestorm. What's different (if anything)? What needs to be added (e.g. for power domain control, sampling the RNG, etc)?

### CODE REVIEW

File: boards/imix/debug.gdb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+target remote :3333
+monitor reset
```
Comment (brghena) [inline boards/imix/debug.gdb]: Do you know why we `monitor reset` multiple times? I've seen many gdb files that do it, but I've never known why.

Comment (alevy) [inline boards/imix/debug.gdb]: actually I copied the debug.gdb file from @shaneleonard.

**Comment by alevy** [comment]
FYI this is programmed with openocd because JLink isn't quite working on Imix yet. However, openocd is actually pretty nice, really easy to set up on newer versions of Linux, open source, and not _terribly_ slow...

**Comment by ppannuto** [comment]
I put about 60s of effort into this, but out-of-the-box, the v0.9 release
of openocd doesn't even compile on my mac:

libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../.. -I../../../src
-I../../../src -I../../../src/helper
-DPKGDATADIR=\"/usr/local/share/openocd\" -DBINDIR=\"/usr/local/bin\"
-I../../../jimtcl -I../../../jimtcl -g -O2 -Wall -Wstrict-prototypes
-Wformat-security -Wshadow -Wextra -Wno-unused-parameter
-Wbad-function-cast -Wcast-align -Wredundant-decls -Werror -MT sim3x.lo -MD
-MP -MF .deps/sim3x.Tpo -c sim3x.c -o sim3x.o
sim3x.c:867:20: error: address of array 'sim3x_info->device_package' will
always evaluate to 'true'
      [-Werror,-Wpointer-bool-conversion]
                        if (sim3x_info->device_package) {
                        ~~  ~~~~~~~~~~~~^~~~~~~~~~~~~~
1 error generated.

This is likely benign and is only an issue b/c clang it the default
compiler and catches some things that gcc doesn't, but the point is that we
may not want to rely on openocd, @brghena and @bradjc were both mumbling
non-niceties about it which prompted me to try.

**Comment by alevy** [comment]
Hrm... interesting... Yeah, I just got it from the Arch Linux repos. @shaneleonard don't you use OS X? Any pointers?

**Comment by alevy** [comment]
In any case, JLinkExe just doesn't work yet with the Imix (probably electrical issues) and openocd does... so for now we're stuck.

**Comment by immesys** [comment]
FYI the saleae has a jtag protocol analyser so you may want to attach it and see why the JLinkExe is failing on the Imix. If it is electrical, you will be able to see the deformed waveforms and can likely fix it by lowering the clock speed.

File: boards/imix/debug.gdb (hunk 1 above)
Comment (shaneleonard) [inline boards/imix/debug.gdb]: Depending on the chip, `monitor reset` can sometimes be flaky, and resetting twice usually guarantees that the board is in a fully defined state.

**Comment by shaneleonard** [comment]
With regard to openocd, I believe I used Homebrew to install it: `brew install open-ocd`. I don't recall having any compilation issues.

**Comment by ppannuto** [comment]
Ah, the brew forumla must have some patches to get things working. Doesn't
work from source, works fine from brew.

(In my 60s of effort I only did a "brew search openocd", which failed b/c
of the missing '-')

**Comment by shaneleonard** [comment]
@immesys Thanks for the tip. I've had mixed success with lowering the clock speed, and I had thought I got it to work by removing a couple of pullup resistors, but then the issue came back without me changing anything. The good news is that at the very least, removing the pull-ups allowed me to dramatically speed up openocd.

**Comment by shaneleonard** [comment]
@ppannuto Glad it worked. Those dashes will get ya

**Pull request merged**